sats-dna/src/main/kotlin/com/sats/dna/components/card/ChallengeCard.kt

```diff
@@ -0,0 +1,395 @@
+        val imageUrl: String?,
+        val title: String,
+        val tagText: String,
+        val progress: Float,
+        val statusText: String,
+        val onCardClick: () -> Unit,
+        val modifier: Modifier = Modifier,
+    ) : ChallengeCardState
+
+    class DisabledChallengeCard(
+        val imageUrl: String?,
+        val title: String,
+        val statusText: String,
+        val onCardClick: () -> Unit,
+        val onDeleteClick: () -> Unit,
+        val modifier: Modifier = Modifier,
+    ) : ChallengeCardState
+}
+
+@Composable
+private fun ChallengeCardLayout(
+    imageUrl: String?,
+    title: String,
+    onCardClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    isEnabled: Boolean = true,
+    subtitle: String? = null,
+    tag: @Composable (() -> Unit?)? = null,
+    deleteButton: @Composable (() -> Unit?)? = null,
+    bottomContent: @Composable (() -> Unit?)? = null,
+) {
+    CompositionLocalProvider(LocalUseMaterial3 provides true) {
+        SatsChallengeBackground(isEnabled = isEnabled, modifier = modifier.height(IntrinsicSize.Min)) {
+            Column(
+                Modifier
+                    .fillMaxSize()
+                    .clickable(onClick = onCardClick),
+                horizontalAlignment = Alignment.CenterHorizontally,
+                verticalArrangement = Arrangement.SpaceBetween,
+            ) {
+                tag?.let {
+                    Box(
+                        Modifier
+                            .fillMaxWidth()
+                            .padding(top = SatsTheme.spacing.xs),
+                        contentAlignment = Alignment.TopStart,
+                    ) {
+                        tag()
+                    }
+                }
+
+                deleteButton?.let {
+                    Box(
+                        Modifier
+                            .fillMaxWidth(),
+                        contentAlignment = Alignment.TopEnd,
+                    ) {
+                        deleteButton()
+                    }
+                }
+                Column(
+                    horizontalAlignment = Alignment.CenterHorizontally,
+                    verticalArrangement = Arrangement.spacedBy(SatsTheme.spacing.xxs, Alignment.CenterVertically),
+                ) {
+                    SubcomposeAsyncImage(
+                        modifier = Modifier
+                            .padding(top = SatsTheme.spacing.m, bottom = SatsTheme.spacing.xxs)
+                            .size(75.dp)
+                            .aspectRatio(1f)
+                            .clip(SatsTheme.shapes.circle),
+                        model = imageUrl,
+                        contentDescription = null,
+                        contentScale = ContentScale.Crop,
+                        loading = { LoadingPlaceholder(Modifier.fillMaxSize()) },
+                        error = { ErrorFallback(contentDescription, Modifier.fillMaxSize()) },
+                    )
```

Comment:
We should probably use the `SatsChallengeBadge` component for challenge images instead of this, as it already has the defaults we've set below.

```kotlin
SatsChallengeBadge(
    imageUrl = imageUrl,
    contentDescription = null,
    modifier = Modifier.size(75.dp),
)
```

It also has support for showing the progress ring around it by passing a `progress: Float?`.

Reply:
Although I guess we never want the progress around the image, since we show that underneath. 😅

Reply:
Yeah, I'll use it without passing progress as of now, then we can remove progress from the whole ChallengeBadge composable eventually if it's not going to be a part of the new Challenges design 👍